jira/server/scio_search/src/main/java/ScioSearchConfigRestPlugin/impl/ScioSearchIssueSecurityMembers.java

```diff
@@ -0,0 +1,134 @@
+import com.atlassian.jira.issue.security.IssueSecurityLevelPermission;
+import ScioSearchConfigRestPlugin.impl.ScioIssueSecurityMembersResponse.IssueSecuritySchemeMemberInfo;
+import ScioSearchConfigRestPlugin.impl.ScioIssueSecurityMembersResponse.JiraPermissionHolderInfo;
+import ScioSearchConfigRestPlugin.impl.ScioIssueSecurityMembersResponse.AtlassianUser;
+import ScioSearchConfigRestPlugin.impl.ScioIssueSecurityMembersResponse.AtlassianGroup;
+import com.atlassian.extras.common.log.Logger;
+import com.atlassian.plugin.spring.scanner.annotation.imports.JiraImport;
+import com.atlassian.sal.api.pluginsettings.PluginSettings;
+import com.atlassian.sal.api.pluginsettings.PluginSettingsFactory;
+import com.atlassian.sal.api.user.UserManager;
+import com.atlassian.sal.api.user.UserProfile;
+import java.util.List;
+import java.util.ArrayList;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Collection;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+@Named
+@Path("/issue_security_scheme_members")
+public class ScioSearchIssueSecurityMembers {
+
+  private static final Logger.Log logger = Logger.getInstance(ScioSearchIssueSecurityMembers.class);
+
+  @JiraImport private final UserManager userManager;
+
+  @JiraImport private final PluginSettingsFactory pluginSettingsFactory;
+
+  @Inject
+  public ScioSearchIssueSecurityMembers(
+      UserManager userManager, PluginSettingsFactory pluginSettingsFactory) {
+    this.userManager = userManager;
+    this.pluginSettingsFactory = pluginSettingsFactory;
+  }
+
+  private void validateUserIsAdmin() {
+    final UserProfile profile = userManager.getRemoteUser();
+    if (profile == null || !userManager.isSystemAdmin(profile.getUserKey())) {
+      throw new UnauthorizedException("Unauthorized");
+    }
+  }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  public ScioIssueSecurityMembersResponse setTarget(@QueryParam("schemeId") String schemeIdStr) {
+    logger.debug(String.format("Received request for getting issuesecuritymembers: %s", schemeIdStr));
+    validateUserIsAdmin();
+    Long schemeId = Long.parseLong(schemeIdStr);
+
+    final IssueSecurityLevelManager issueSecurityLevelManager = ComponentAccessor.getIssueSecurityLevelManager();
+    final IssueSecuritySchemeManager issueSecuritySchemeManager = ComponentAccessor.getComponentOfType(IssueSecuritySchemeManager.class);
+    final Collection<IssueSecurityLevel> issueSecurityLevels = issueSecurityLevelManager.getIssueSecurityLevels(schemeId);
+
+    final ScioIssueSecurityMembersResponse response = new ScioIssueSecurityMembersResponse();
+    response.startAt = 0;
+    response.isLast = true; // no response pagination
+    if (issueSecurityLevels != null) {
```

Comment:
Prefer early returns to avoid deep nesting. Deep nesting is hard to read, because it's hard to match the braces.

For example:
```
if (issueSecurityLevels == null) {
  return response;
}
for (isl : issueSecurityLevel) {
  if (issueSecurityLevelPermissions == null) {
    continue;
  }
  ...
}
```

Reply:
Done.